include/mxnet/tuple.h

```diff
@@ -475,11 +536,12 @@ class TShape : public Tuple<dim_t> {
    */
   inline mshadow::Shape<2> FlatTo2D(void) const {
     mshadow::Shape<2> s;
-    if (ndim() == 0) return mshadow::Shape2(0, 0);
+    CHECK_GE(ndim(), 0);
```

Comment:
error message

Reply:
or should we use ndim_is_known for the check?

Reply:
Done.